graphql/schemabuilder/pagination.go

```diff
@@ -492,6 +552,121 @@ func getFuncReturnType(fn interface{}) reflect.Type {
 	return typ.Out(0)
 }
 
+func supportedSort(typ reflect.Type) bool {
+	switch typ.Kind() {
+	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64,
+		reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64,
+		reflect.Float32, reflect.Float64,
+		reflect.String:
+		return true
+	}
+	return false
+}
+
+type SortOrder int64
+
+const (
+	SortOrder_Ascending SortOrder = iota
+	SortOrder_Descending
+)
+
+type sortReference struct {
+	index int
+	value interface{}
```

Comment:
can we put the `reflect.Value` directly here?

Reply:
Done